packages/ra-core/src/routing/useGetPathForRecordCallback.ts

```diff
@@ -23,46 +26,76 @@ export const useGetPathForRecordCallback = <
                 );
             }
             const resourceDefinition = resourceDefinitions[finalResource] ?? {};
-            const linkFunc = typeof link === 'function' ? link : () => link;
-
-            const defaultLink = resourceDefinition.hasShow
-                ? 'show'
-                : resourceDefinition.hasEdit
-                  ? 'edit'
-                  : false;
-
-            const isLinkFalse =
-                link === false || (link == null && defaultLink === false);
 
-            if (record == null || isLinkFalse) {
+            if (record == null || link === false) {
                 return false;
             }
-            const linkResult = linkFunc(record, finalResource) ?? defaultLink;
-            const linkResultIsPromise = isPromise(linkResult);
 
-            if (linkResultIsPromise) {
-                return linkResult.then(resolvedLink => {
-                    if (resolvedLink === false) {
-                        // already set to false by default
-                        return;
-                    }
+            // When the link prop is not provided, we infer a default value and check whether users
+            // can access it
+            if (link == null) {
+                const [canAccessShow, canAccessEdit] = await Promise.all([
+                    resourceDefinition.hasShow
```

Comment:
Why do you test `resourceDefinition.hasShow` here and also L54?

Reply:
Because you many not have an authProvider and in this case, useCanAccess will always return true